client/cherami/client.go

```diff
@@ -344,11 +355,16 @@ func getDefaultOptions() *ClientOptions {
 // verifyOptions is used to verify if we have a metrics reporter and
 // a logger. If not, just setup a default logger and a null reporter
 // it also validate the timeout is sane
-func verifyOptions(opts *ClientOptions) *ClientOptions{
+func verifyOptions(opts *ClientOptions) (*ClientOptions, error){
 	if opts == nil {
 		opts = getDefaultOptions()
 	}
-	common.ValidateTimeout(opts.Timeout)
+	if opts.Timeout.Nanoseconds() == 0 {
```

Comment:
You can just do `opts.Timeout == 0` because `time.Duration` is just an alias for `int64`.